sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs

```diff
@@ -34,15 +34,25 @@ public TextSource(string text, string voiceName)
 
         /// <summary> Initializes a new instance of TextSource. </summary>
         /// <param name="text"> Text for the cognitive service to be played. </param>
-        /// <param name="sourceLocale"> The culture info of the voice. </param>
+        /// <param name="sourceLocale"> The culture info string of the voice. </param>
         /// <param name="gender"> The gender of the voice. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="text"/> is null. </exception>
-        public TextSource(string text, CultureInfo sourceLocale, GenderType gender)
+        /// <exception cref="ArgumentException"> <paramref name="sourceLocale"/> is wrong format. </exception>
+        public TextSource(string text, string sourceLocale, GenderType gender)
         {
             Argument.AssertNotNull(text, nameof(text));
 
+            try
```

Comment:
@wangrui-msft  Should we add the same check for SpeechLanguage?

Reply:
The speech language doesn't have the constructor, do you suggest to add a new constructor for speech language?